src/coreclr/src/vm/dwreport.cpp

```diff
@@ -189,14 +189,17 @@ HRESULT DwGetFileVersionInfo(
     CONTRACTL
     {
         NOTHROW;
-        GC_NOTRIGGER;
+        GC_TRIGGERS;
         MODE_ANY;
     }
     CONTRACTL_END;
 
     major = minor = build = revision = 0;
     ULARGE_INTEGER appVersion = { 0, 0 };
 
+    // Preempt to let GC suspend
+    GCX_PREEMP();
+
     HRESULT result = GetFileVersion(wszFilePath, &appVersion);
```

Comment:
It would be nice to change the contract on `GetFileVersion` from `MODE_ANY` to `MODE_PREEMP` to make sure that it won't ever be called in cooperative mode again.